.gitignore

```diff
@@ -1,16 +1,5 @@
+Cargo.lock
 target/*
+.DS_Store
 *__pycache__*
-*.mypy_cache*
-*.egg-info*
-*.coverage
-*/docs/generated*
-*coveragehtml/*
-*/docs/_build*
-*.DS_Store
-*.ipynb_checkpoints*
-*.json
-*.yaml
-.vscode/*
-*.profraw
-*.lcov
-*code-workspace
```

Comment:
code-workspace and .vscode can be created by VS code we should keep both of those